src/pow.cpp

```diff
@@ -14,18 +14,14 @@
 #include "uint256.h"
 #include "util.h"
 
+#define NUM_OF_BLOCKS_DIFF_ADJ 10
 unsigned int GetNextWorkRequired(const CBlockIndex* pindexLast, const CBlockHeader *pblock, const Consensus::Params& params)
 {
     assert(pindexLast != nullptr);
     unsigned int nProofOfWorkLimit = UintToArith256(params.powLimit).GetCompact();
 
     int nHeightNext = pindexLast->nHeight + 1;
-    if (nHeightNext >= params.BTGHeight  && nHeightNext < params.BTGHeight + params.BTGPremineWindow)
-    {
-        // Lowest difficulty for Bitcoin GPU premining period.
-        return nProofOfWorkLimit;
-    }
-    else if (nHeightNext % params.DifficultyAdjustmentInterval() != 0)
+ if (nHeightNext % params.DifficultyAdjustmentInterval() != 0)
```

Comment:
Do you want to calculate the difficulty by a sliding window?

Reply:
No, I have added a new params for the blockchain consensus that is calculating the difficulty, each block.